### PR TITLE
Add more support with Spring for Apache Kafka

### PR DESCRIPTION
**Pull request by artembilan**
* Add missed `JsonDeserializer` into type hints
* Add `@InitializationHint` for some `spring-kafka` JSON components
which need to initialize their `static` props for `ClassUtils.isPresent()`
* Improve `kafka` sample for JSON data interaction
* Add a `@TypeHint` for end-user data classes used in the sample
* Improve `docker-compose.yml` for extra image of the current `kafka` sample

### CODE REVIEW

**Comment by artembilan** [comment]
Depends on: https://github.com/spring-projects/spring-kafka/pull/1881

File: samples/kafka/src/main/java/com/example/kafka/KafkaApplication.java [hunk 1]
```diff
@@ -29,15 +34,38 @@ public NewTopic topic() {
...
+			ConsumerFactory<String, Greeting> cf) {
+
+		cf.addListener(new ConsumerFactory.Listener() {
 
-		cf.addListener(new ConsumerFactory.Listener() { });
+		});
 		return args -> {
-			template.send("graal", "foo");
+			template.send("graal", new Greeting("Hello from GraalVM!"));
 			System.out.println("++++++Sent:foo");
```
Comment (garyrussell) [inline samples/kafka/src/main/java/com/example/kafka/KafkaApplication.java:45]: Change this to match?

**Review by garyrussell — CHANGES_REQUESTED** [review]

**Comment by garyrussell** [comment]
Didn't you test it?

You need to update the verify.sh https://github.com/spring-projects-experimental/spring-native/blob/main/samples/kafka/verify.sh

**Comment by artembilan** [comment]
Huh? What is that? Never used those `.sh`. Tried `build.sh` once, but it was silent and in the end it produces an `.exe` which is not always working on Windows.
Yeah.. Thanks for the pointer! Let me see what that `verify.sh` does for us and fix it respectively! 
Docs don't mention this file though...

**Comment by garyrussell** [comment]
See https://github.com/spring-projects-experimental/spring-native/blob/main/scripts/test.sh

called from https://github.com/spring-projects-experimental/spring-native/blob/main/samples/kafka/build.sh

**Comment by artembilan** [comment]
Repushed after rebasing to `main`.
Checked locally with those `build.sh` and `verify.sh`: works well even on Windows for me ! 😄

**Comment by garyrussell** [comment]
Hmmm - on MacOS:

```
***************************
APPLICATION FAILED TO START
***************************

Description:

Native reflection configuration for org.springframework.kafka.support.serializer.JsonDeserializer is missing.
```

**Comment by artembilan** [comment]
Correct. You must build and install locally the main project first.
That `JsonDeserializer` was missed in the original `KafkaHints`, so I added it as a part of my PR.

**Comment by garyrussell** [comment]
Ah; ok; sorry.

**Review by garyrussell — APPROVED** [review]
LGTM; however since it needs a spring-kafka snapshot, it can't be merged unless there are no plans for a release before August 16 (when spring-kafka 2.7.5 will be released). But I could consider releasing it earlier than that, if needed.

**Comment by artembilan** [comment]
Well, there is no strong dependency on spring-kafka in the main project. Right now it is only a matter of a sample. `spring-native` can be released as is. 
We may find something else though for better compatibility before that August 16th...

**Pull request closed without merging**